src/main/java/org/privacyidea/PrivacyIDEA.java

```diff
@@ -150,6 +157,11 @@ private PIResponse getPIResponse(String type, String input, String pass, Map<Str
             params.put(TRANSACTION_ID, transactionID);
         }
         String response = runRequestAsync(ENDPOINT_VALIDATE_CHECK, params, headers, false, POST);
+        // Shutdown the scheduler if user successfully authenticated
+        if (this.parser.parsePIResponse(response) != null && this.parser.parsePIResponse(response).value)
```

Comment:
do not call `parsePIResponse` twice

Reply:
right
[faf235b](https://github.com/privacyidea/java-client/pull/70/commits/faf235b40b1d76be06c097ad3f717e26213bc47b)